classifier/run.py

```diff
@@ -0,0 +1,263 @@
+
+'''
+Script to train and test litter classifier.
+'''
+import torch
+import torchvision
+import torchvision.transforms as transforms
+
+import torch.nn as nn
+import torch.optim as optim
+from torch.utils.data.sampler import SubsetRandomSampler
+from torch.autograd import Variable
+
+import matplotlib.pyplot as plt
+import numpy as np
+import os
+import argparse
+
+
+def get_args_parser():
+    parser = argparse.ArgumentParser(
+        'Train and test network for classification task')
+    parser.add_argument('--data_img',
+                        help='path to directory with subdirectories with images',
+                        type=str)
+    parser.add_argument('--out',
+                        help='path to main directory with checkpoints',
+                        type=str)
+    parser.add_argument('--mode', default='train',
+                        help='type of procedure: test or train',
+                        choices=['train', 'test'],
+                        type=str)
+    parser.add_argument('--name', default='test.png',
+                        help='path to save test images', type=str)
+    parser.add_argument('--num', help='number of images to display',
+                        default=5, type=int)
+    parser.add_argument('--device', help='specify device to use',
+                        default="cuda:0", type=str)
+    parser.add_argument('--epochs', default=30, type=int)
+    parser.add_argument('--lr', default=0.001, type=float)
+    parser.add_argument('--momentum', default=0.9, type=float)
+    return parser
+
+
+def make_weights_for_balanced_classes(images, nclasses=6):
+    count = [0] * nclasses
+    for item in images:
+        count[item[1]] += 1
+    weight_per_class = [0.] * nclasses
+    N = float(sum(count))
+    for i in range(nclasses):
+        weight_per_class[i] = N / float(count[i]) if count[i] != 0 else N
+    weight = [0] * len(images)
+    for idx, val in enumerate(images):
+        weight[idx] = weight_per_class[val[1]]
+    return weight
+
+
+def load_split_train_test(datadir, resize_img=(224, 224),
```

Comment:
pytorch have a special module "dataset" and "dataloader" easy to use, consider using it in future

Reply:
Yea, I used this inside this function. I want to prepare data split in two manners - completely random and based on WeigthedRandomSampler to tackle imbalanced class problem.